ldaptor/protocols/ldap/ldapserver.py

```diff
@@ -62,10 +62,10 @@ def connectionLost(self, reason=protocol.connectionDone):
         """Called when TCP connection has been lost"""
         self.connected = 0
 
-    def queue(self, id, op):
+    def queue(self, id, op, controls):
         if not self.connected:
             raise LDAPServerConnectionLostException()
-        msg = pureldap.LDAPMessage(op, id=id)
+        msg = pureldap.LDAPMessage(op, controls=controls, id=id)
```

Comment:
controls=None is handled in pureldap so no need to handle it here. just pass it.